jinahub/indexers/searcher/HnswlibSearcher/hnswlib_searcher.py

```diff
@@ -131,7 +134,12 @@ def search(
         for i, (indices_i, dists_i) in enumerate(zip(indices, dists)):
             for idx, dist in zip(indices_i, dists_i):
                 match = Document(id=self._ids_to_inds.inverse[idx])
-                match.scores[self.metric] = dist
+                if self.is_distance:
+                    match.scores[self.metric] = dist
+                elif self.metric in ["inner_product", "cosine"]:
+                    match.scores[self.metric] = 1 - dist
+                else:
+                    match.scores[self.metric] = 1 / (1 + dist)
```

Comment:
Not needed

Reply:
Please change this so that the logic here is explicit - `else` should be `elif `self.metric == "euclidean"`